lib/OpenQA/JobSettings.pm

```diff
@@ -104,4 +111,36 @@ sub handle_plus_in_settings {
     }
 }
 
+# Given a hashref of settings, parse any whose names end in _URL
+# to the short name, then if there is not already a setting with
+# the short name, set it to the filename from the URL (with the
+# compression extension removed in the case of _DECOMPRESS_URL).
+# This has to happen *before* generate_jobs
+sub parse_url_settings {
+    my ($settings) = @_;
+    for my $setting (keys %$settings) {
+        my ($short, $do_extract) = get_url_short($setting);
+        next unless ($short);
+        next if defined($settings->{$short});
+        # As this comes in from an API call, URL will be URI-encoded
+        # This obviously creates a vuln if untrusted users can POST
+        $settings->{$setting} = url_unescape($settings->{$setting});
+        my $url = $settings->{$setting};
+        my $filename;
+        $filename = Mojo::URL->new($url)->path->parts->[-1];
+        if ($do_extract) {
+            # if user wants to extract downloaded file, final filename
+            # will have last extension removed
+            $filename = fileparse($filename, qr/\.[^.]*/);
+        }
+        $settings->{$short} = $filename;
+        if (!$settings->{$short}) {
+            log_debug("Unable to get filename from $url. Ignoring $setting");
+            delete $settings->{$short} unless $settings->{$short};
```

Comment:
is the `unless` duplicated here? we already have the `if (!$settings->{$short})` above.

Reply:
it is, but it was in the original code too; I was trying to keep it as intact as possible while splitting it up. I can poke a bit and see if it was there for some kind of odd reason.

Reply:
OK, it looks like the redundancy was introduced in f54a18fef84da4fe72d03ca7d12cbc37db705ae5 , which did this:

             if (!$args{$short}) {
                 $args{$short} = Mojo::URL->new($url)->path->parts->[-1];
    -            delete $args{$short} unless $args{$short};
    +            if (!$args{$short}) {
    +                OpenQA::Utils::log_warning("Unable to get filename from $url. Ignoring $arg");
    +                delete $args{$short} unless $args{$short};
    +                next;
    +            }
             }

which looks like it was just an oversight, so it's probably okay to fix it. I'm going to make it a separate commit, though.